elm-format-lib/src/Parse/Helpers.hs

```diff
@@ -1,283 +1,255 @@
 qualifiedTag elmVersion =
-    TagRef
-        <$> many (try $ const <$> capVar elmVersion <*> string ".")
-        <*> capVar elmVersion
-
+  TagRef
+    <$> many (try $ const <$> capVar elmVersion <*> string ".")
+    <*> capVar elmVersion
 
 rLabel :: ElmVersion -> IParser LowercaseIdentifier
 rLabel = lowVar
 
-
 innerVarChar :: ElmVersion -> IParser Char
 innerVarChar elmVersion =
-    if syntax_0_19_disallowApostropheInVars elmVersion
-        then alphaNum <|> char '_' <?> "more letters in this name"
-        else alphaNum <|> char '_' <|> char '\'' <?> "more letters in this name"
-
+  if syntax_0_19_disallowApostropheInVars elmVersion
+    then alphaNum <|> char '_' <?> "more letters in this name"
+    else alphaNum <|> char '_' <|> char '\'' <?> "more letters in this name"
 
 makeVar :: ElmVersion -> IParser Char -> IParser String
 makeVar elmVersion firstChar =
-  do  variable <- (:) <$> firstChar <*> many (innerVarChar elmVersion)
-      if variable `elem` reserveds
-        then parserFail $ parseError (Message (Syntax.keyword variable))
-        else return variable
-
+  do
+    variable <- (:) <$> firstChar <*> many (innerVarChar elmVersion)
+    if variable `elem` reserveds
+      then parserFail $ parseError (Message (Syntax.keyword variable))
+      else return variable
 
 reserved :: ElmVersion -> String -> IParser ()
 reserved elmVersion word =
   expecting ("reserved word `" ++ word ++ "`") $
-    do  _ <- string word
-        notFollowedBy (innerVarChar elmVersion)
-        return ()
-
+    do
+      _ <- string word
+      notFollowedBy (innerVarChar elmVersion)
+      return ()
 
 -- INFIX OPERATORS
 
 anyOp :: ElmVersion -> IParser (Ref [UppercaseIdentifier])
 anyOp elmVersion =
   (betwixt '`' '`' (qualifiedVar elmVersion) <?> "an infix operator like `andThen`")
-  <|> (OpRef <$> symOp)
-
+    <|> (OpRef <$> symOp)
 
 symOp :: IParser SymbolIdentifier
 symOp =
-  do  op <- many1 (satisfy Help.isSymbol) <?> "an infix operator like +"
-      guard (op `notElem` [ "=", "..", "->", "--", "|", "\8594", ":" ])
-      case op of
-        "." -> notFollowedBy lower >> return (SymbolIdentifier op)
-        _   -> return $ SymbolIdentifier op
-
+  do
+    op <- many1 (satisfy Help.isSymbol) <?> "an infix operator like +"
+    guard (op `notElem` ["=", "..", "->", "--", "|", "\8594", ":"])
+    case op of
+      "." -> notFollowedBy lower >> return (SymbolIdentifier op)
+      _ -> return $ SymbolIdentifier op
 
 symOpInParens :: IParser SymbolIdentifier
 symOpInParens =
-    parens' symOp
-
+  parens' symOp
 
 -- COMMON SYMBOLS
 
 equals :: IParser ()
 equals =
-  const () <$> char '=' <?> "="
-
+  void (char '=' <?> "=")
 
 lenientEquals :: IParser ()
 lenientEquals =
-  const () <$> (char '=' <|> char ':') <?> "="
-
+  void ((char '=' <|> char ':') <?> "=")
 
 rightArrow :: IParser ()
 rightArrow =
-  const () <$> (string "->" <|> string "\8594") <?> "->"
-
+  void (string "->" <|> string "\8594" <|> string "=>") <?> "->"
```

Comment:
This is the only relevant bit to this PR ☝🏻

Reply:
I'm not the maintainer, but if this is the only relevant part of the PR, I think this should be the only thing in the PR (well, with the tests).
Otherwise it's a lot of noise. Also, some of this *might* be taken straight from the Elm compiler, and we'd want to keep that as close as possible to the original?

Reply:
> some of this might be taken straight from the Elm compiler, and we'd want to keep that as close as possible to the original?

That's not really a concern any more, since elm-format's still forked from Elm 0.16's compiler, which was completely re-written in Elm 0.19.  So there's really no chance of of easily merging parser changes anymore until https://github.com/avh4/elm-format/issues/301 is finished.

Reply:
Done, sorry for the noise!